packages/atomic/src/components/search/result-template-components/atomic-result-image/atomic-result-image.tsx

```diff
@@ -36,27 +41,56 @@ export class AtomicResultImage implements InitializableComponent {
     return Array.isArray(value) ? value[0] : value;
   }
 
-  public render() {
-    const url = this.url;
+  private handleImageError() {
+    const currentImageUrl = this.host.querySelector('img')!.src;
+    let message = `The image url "${currentImageUrl}" is not valid or could not be loaded.`;
 
-    if (!url) {
-      this.host.remove();
-      return;
+    if (this.fallback) {
+      this.host.querySelector('img')!.src = this.fallback;
+    } else {
+      message += ' You might want to add a "fallback" property.';
     }
+    this.bindings.engine.logger.warn(message, this.host);
+  }
+
+  private handleInvalidUrl(message: string, fallback: string | undefined) {
+    let errorMessage = message;
 
-    if (typeof url !== 'string') {
-      this.bindings.engine.logger.error(
-        `Expected "${this.field}" to be a text field.`,
-        this.host
-      );
+    if (fallback) {
+      errorMessage += ` The fallback "${fallback}" is being used.`;
+      this.bindings.engine.logger.warn(errorMessage, this.host);
+      return fallback;
+    } else {
+      errorMessage += ' You might want to add a "fallback" property.';
+      this.bindings.engine.logger.error(errorMessage, this.host);
       this.host.remove();
-      return;
+      return null;
     }
+  }
+
+  public render() {
+    let url = this.url;
 
+    if (!url) {
+      const message = `"${this.field}" is missing. Please review the indexing.`;
+      url = this.handleInvalidUrl(message, this.fallback);
+      if (!url) {
+        return;
+      }
+    }
+
+    if (url && typeof url !== 'string') {
+      const message = `Expected "${this.field}" to be a text field. Please review the indexing.`;
+      url = this.handleInvalidUrl(message, this.fallback);
+      if (!url) {
+        return;
+      }
+    }
```

Comment:
I don't think we need this many errors. I think a warning to tell the user "hey, there's a fallback feature you can use" is handy, but beyond that, I think errors/warnings would be redundant and would pollute the logs.

I also prefer not to concatenate messages over different scopes, I think it makes it more difficiult to interpret the possible errors.